lib/readability/published_at_finder.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Readability.PublishedAtFinder do
+  @moduledoc """
+  Extract the published at.
+  """
+
+  @type html_tree :: tuple | list
+
+  @strategies [:meta_tag, :time_element, :data_attribute, :text]
+
+  @doc """
+  Extract the published at.
+  """
+  @spec find(html_tree) :: %DateTime{} | %Date{} | nil
+  def find(html_tree) do
+    value =
+      Enum.find_value(@strategies, fn strategy ->
+        strategy(strategy, html_tree)
+      end)
+
+    if value do
+      parse(value)
+    end
+  end
+
+  defp strategy(:meta_tag, html_tree) do
+    selector = "meta[property='article:published_time'], meta[property='article:published']"
+
+    html_tree
+    |> Floki.find(selector)
+    |> Enum.flat_map(&Floki.attribute(&1, "content"))
```

Comment:
Floki.find + flat_map(&Floki.attribute/2) = [Floki.attribute/3 ](https://hexdocs.pm/floki/Floki.html#attribute/3). This should simplify these cases.

Reply:
Ah nice 🙏 I didn't realise I could do that! I've pushed a fix for this.